### PR TITLE
T0 19 new arrivals

### PR DESCRIPTION
**Pull request by AlaaYa7ia**
New arrivals should be any products that have been created for 3 months for example if you are in April, any products that were created in January will be new arrivals.
------

1- Routs: 
------
**GET /api/products/new-arrivals - Get new arrivals (with pagination)**

POST /api/products - Add a new product.

PUT /api/products/:id - Update product details (e.g., price, stock, etc.).

DELETE /api/products/:id - Delete a product.

+ Validation

2- productsController
------
3- productRepository
------
4- productService
------
**5- updated the product model (I'm not sure if I need it without the testing)**
------

**Note: unit tests will be in another PR.**

### CODE REVIEW

File: src/controllers/userController.ts [hunk 1]
```diff
@@ -1,38 +1,67 @@
-import { Request, Response } from 'express';
-import { validationResult } from 'express-validator';
-import { registerUser } from '../services/userService';
-import { verifyPassword } from '../services/userService';
+import { Request, Response } from "express";
```
Comment (AlaaYa7ia) [inline src/controllers/userController.ts:1]: I didn't change anything here, it's only automatic formating.

File: src/controllers/userController.ts [hunk 2]
```diff
@@ -1,38 +1,67 @@
-import { Request, Response } from 'express';
-import { validationResult } from 'express-validator';
-import { registerUser } from '../services/userService';
-import { verifyPassword } from '../services/userService';
```
Comment (AlaaYa7ia) [inline src/controllers/userController.ts:4]: added these two lines together only,  I didn't change the rest of the file, it's only automatic formatting.

**Review by batooldshilleh — APPROVED** [review]
Great job!

**Comment by Lojain-Abdalrazaq** [comment]
Great job, Alaa! 🙏 I noticed that the `Product` model is missing the image URL property. The product image should be stored in Firebase Storage.

**Review by Lojain-Abdalrazaq — CHANGES_REQUESTED** [review]
Great job, Alaa!!! 🙏 I noticed that the `Product` model is missing the image URL property. The product image should be stored in Firebase Storage, and the URL should represent the image URL in Firebase.

**Pull request merged**